android/app/src/main/java/org/fossasia/openevent/viewmodels/SearchActivityViewModel.java

```diff
@@ -0,0 +1,81 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.realm.RealmResults;
+
+/**
+ * Created by sanyog on 1/1/18.
+ */
+
+public class SearchActivityViewModel extends ViewModel {
+
+
+    private MutableLiveData<List<Object>> searchResults;
+    private RealmDataRepository realmRepo;
+
+    private String searchText = "";
+
+    public SearchActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<List<Object>> getSearchResults() {
+        RealmResults<Microlocation> realmResults_microloc;
+        RealmResults<Track> realmResults_track;
+        RealmResults<Session> realmResults_session;
+        RealmResults<Speaker> realmResults_speaker;
+        List<Object> objectList = new ArrayList<>();
+
+        if (searchResults == null) {
+            searchResults = new MutableLiveData<>();
+            realmResults_track = realmRepo.getTracks();
+            objectList.add(realmResults_track);
+
+            int a = realmResults_track.size();
+            realmResults_session =realmRepo.getSessions();
+            objectList.add(realmResults_session);
+
+            int b = realmResults_session.size();
+            realmResults_speaker = realmRepo.getSpeakers();
+            objectList.add(realmResults_speaker);
+
+            int c = realmResults_speaker.size();
+            realmResults_microloc = realmRepo.getLocations();
+            objectList.add(realmResults_microloc);
+
+            int d = realmResults_track.size();
+            if(a+b+c+d == 0){
+                return null;
+            }
+            else {
+                 searchResults.setValue(objectList);
+            }
+        }
+        return searchResults;
+    }
+
+    public String getSearchText() {
+        return searchText;
+    }
+
+    public void setSearchText(String searchText) {
+        this.searchText = searchText;
+    }
+
+    @Override
+    protected void onCleared() {
+        super.onCleared();
```

Comment:
You should dispose all the changeListeners if you have any, otherwise this method is redundant since it is only calling it's super class.